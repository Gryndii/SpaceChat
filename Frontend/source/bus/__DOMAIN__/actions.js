//Types
import { types } from './types';

export const actions = {
    action: () => {
        return {
            type: types.TYPE,
        };
    },
};
