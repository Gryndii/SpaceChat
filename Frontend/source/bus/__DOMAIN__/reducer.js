//Types
import { types } from './types';

const initialState = {};

export const reducer = (state = initialState, action) => {
    switch (action.type) {
        case types.TYPE:
            return state;

        default:
            return state;
    }
};
