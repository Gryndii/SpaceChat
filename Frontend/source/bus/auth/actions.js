//Types
import { types } from './types';

export const signupAsync = (userData) => {
  return {
    type: types.SIGNUP_ASYNC,
    payload: userData,
  };
};

export const loginAsync = (credentials) => {
  return {
    type: types.LOGIN_ASYNC,
    payload: credentials,
  };
};

export const authenticateAsync = () => {
  return {
    type: types.AUTHENTICATE_ASYNC,
  };
};

export const initializeAsync = () => {
  return {
    type: types.INITIALIZE_ASYNC,
  };
};

export const logoutAsync = () => {
  return {
    type: types.LOGOUT_ASYNC,
  };
};

export const authenticate = () => {
  return {
    type: types.AUTHENTICATE,
  };
};

export const initialize = () => {
  return {
    type: types.INITIALIZE,
  };
};

export const logout = () => {
  return {
    type: types.LOGOUT,
  };
};