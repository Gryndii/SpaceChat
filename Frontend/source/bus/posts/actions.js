//Types
import { types } from './types';

export const fetchPostsAsync = () => {
  return {
    type: types.FETCH_POSTS_ASYNC,
  };
};

export const likePostAsync = (postId) => {
  return {
    type: types.LIKE_POST_ASYNC,
    payload: postId,
  };
};

export const unlikePostAsync = (postId) => {
  return {
    type: types.UNLIKE_POST_ASYNC,
    payload: postId,
  };
};

export const createPostAsync = (comment) => {
  return {
    type: types.CREATE_POST_ASYNC,
    payload: comment,
  };
};

export const removePostAsync = (postId) => {
  return {
    type: types.REMOVE_POST_ASYNC,
    payload: postId,
  };
};

export const fillPosts = (posts) => {
  return {
    type: types.FILL_POSTS,
    payload: posts,
  };
};

export const likePost = (likePostData) => {
  return {
    type: types.LIKE_POST,
    payload: likePostData,
  };
};

export const unlikePost = (unlikePostData) => {
  return {
    type: types.UNLIKE_POST,
    payload: unlikePostData,
  };
};

export const createPost = (post) => {
  return {
    type: types.CREATE_POST,
    payload: post,
  };
};

export const removePost = (postId) => {
  return {
    type: types.REMOVE_POST,
    payload: postId,
  };
};