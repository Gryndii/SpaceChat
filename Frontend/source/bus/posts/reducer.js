//Core
import { List, fromJS } from 'immutable';

//Types
import { types } from './types';

const initialState = List();

export const postsReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.FILL_POSTS: {
      return fromJS(action.payload);
    }

    case types.CREATE_POST: {
      return state.unshift(fromJS(action.payload));
    }

    case types.REMOVE_POST: {
      return state.filter(post => post.get('id') !== action.payload);
    }

    case types.LIKE_POST: {
      const index = state.findLastIndex(post => post.get('id') === action.payload.postId);

      return state.updateIn([index, 'likes'], likes => likes.unshift(action.payload.liker));
    }

    case types.UNLIKE_POST: {
      const index = state.findIndex(post => post.get('id') === action.payload.postId);
    

      return state.updateIn([index, 'likes'], likes => {
        return likes.filter(like => like.get('id') !== action.payload.userId)
      });
    }

    default: {
      return state;
    }
  };
};