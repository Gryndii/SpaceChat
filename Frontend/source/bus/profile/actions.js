//Types
import { types } from './types';

export const fillProfile = (profile) => {
  return {
    type: types.FILL_PROFILE,
    payload: profile,
  };
};

export const clearProfile = () => {
  return {
    type: types.CLEAR_PROFILE,
  };
};