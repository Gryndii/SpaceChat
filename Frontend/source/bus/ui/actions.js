//Types
import { types } from './types';

export const startFetching = () => {
  return {
    type: types.START_FETCHING,
  };
};

export const stopFetching = () => {
  return {
    type: types.STOP_FETCHING,
  };
};

export const setOnlineState = () => {
  return {
    type: types.SET_ONLINE_STATE,
  };
};

export const setOflineState = () => {
  return {
    type: types.SET_OFLINE_STATE,
  };
};