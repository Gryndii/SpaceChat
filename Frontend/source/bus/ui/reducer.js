//Core
import { fromJS, Map } from 'immutable';

//Types
import { types } from './types';

const initialState = Map({
    isFetching: false,
    alertPopup: {
        isOpened: false,
        title:    '',
        message:  '',
    },
});

export const uiReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.START_FETCHING:
            return state.set('isFetching', true);

        case types.STOP_FETCHING:
            return state.set('isFetching', false);

        case types.OPEN_ALERT_POPUP:
            return state.setIn([ 'alertPopup' ], fromJS({
                isOpened: true,
                title:    action.payload.title,
                message:  action.payload.message,
            }));

        case types.CLOSE_ALERT_POPUP:
            return state.setIn([ 'alertPopup' ], fromJS({
                isOpened: false,
                title:    '',
                message:  '',
            }));

        default:
            return state;
    }
};
