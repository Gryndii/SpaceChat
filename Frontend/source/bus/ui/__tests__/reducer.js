//Reducer
import { uiReducer } from '../reducer';

//Actions
import * as uiActions from '../actions';

describe('ui reducer:', () => {
  test('should return state by default', () => {
    expect(uiReducer(void 0, {type: 'TEST_ACTION'})).toMatchSnapshot()
  });

  test('should toggle isFetching to true', () => {
    expect(uiReducer(void 0, {type: 'START_FETCHING'})).toMatchSnapshot()
  });

  test('should toggle isFetching to false', () => {
    expect(uiReducer(void 0, {type: 'STOP_FETCHING'})).toMatchSnapshot()
  });

  test('should toggle isOnline to true', () => {
    expect(uiReducer(void 0, {type: 'SET_ONLINE_STATE'})).toMatchSnapshot()
  });

  test('should toggle isOnline to false', () => {
    expect(uiReducer(void 0, {type: 'SET_OFLINE_STATE'})).toMatchSnapshot()
  });
});