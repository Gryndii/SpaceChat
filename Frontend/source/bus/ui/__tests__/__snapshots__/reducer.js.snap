// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui reducer: should return state by default 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`ui reducer: should toggle isFetching to false 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`ui reducer: should toggle isFetching to true 1`] = `
Immutable.Map {
  "isFetching": true,
  "isOnline": false,
}
`;

exports[`ui reducer: should toggle isOnline to false 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`ui reducer: should toggle isOnline to true 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": true,
}
`;
